src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -223,5 +223,378 @@ function (indicator_max::IndicatorMax)(u::AbstractArray{<:Any,4},
+      network_input = network_input ./ max(maximum(abs.(network_input)),1)
+      probability_troubled_cell = network(network_input)[1]
+      if alpha_continuous && !alpha_amr
+        # Set good cells to 0 and troubled cells to continuous value of the network prediction
+        if probability_troubled_cell > 0.5
+          alpha_element = probability_troubled_cell
+        else
+          alpha_element = 0
+        end
+
+        # Take care of the case close to pure FV
+        if alpha_element > 1 - alpha_min
+        alpha_element = one(alpha_element)
+        end
+
+        # Clip the maximum amount of FV allowed
+        alpha[element] = alpha_max * alpha_element
+      elseif !alpha_continuous && !alpha_amr
+        # Set good cells to 0 and troubled cells to 1
+        if probability_troubled_cell > 0.5
+          alpha[element] = 1
+        else
+          alpha[element] = 0
+        end
+      elseif alpha_amr
+        # The entire continuous output of the neural network is used for AMR
+        alpha_element = probability_troubled_cell
+      end
+
+    end
+  elseif indicator_type == "NNRH"
+    @unpack alpha, alpha_tmp, indicator_threaded, modal_threaded, modal_tmp1_threaded = indicator_ann.cache #X, network_input
+    # TODO: Taal refactor, when to `resize!` stuff changed possibly by AMR?
+    #       Shall we implement `resize!(semi::AbstractSemidiscretization, new_size)`
+    #       or just `resize!` whenever we call the relevant methods as we do now?
+    resize!(alpha, nelements(dg, cache))
+    if alpha_smooth
+      resize!(alpha_tmp, nelements(dg, cache))
+    end
+
+    c2e = zeros(Int, length(mesh.tree))
+    for element in eachelement(dg, cache)
+      c2e[cache.elements.cell_ids[element]] = element
+    end
+
+    X = Array{Float64}(undef, 3, nelements(dg, cache))
+
+    @threaded for element in eachelement(dg, cache)
+      indicator  = indicator_threaded[Threads.threadid()]
+      modal      = modal_threaded[Threads.threadid()]
+      modal_tmp1 = modal_tmp1_threaded[Threads.threadid()]
+
+      # Calculate indicator variables at Gauss-Lobatto nodes
+      for j in eachnode(dg), i in eachnode(dg)
+        u_local = get_node_vars(u, equations, dg, i, j, element)
+        indicator[i, j] = indicator_ann.variable(u_local, equations)
+      end
+
+      # Convert to modal representation
+      multiply_scalar_dimensionwise!(modal, dg.basis.inverse_vandermonde_legendre, indicator, modal_tmp1)
+      # Save linear modal coefficients for the network input
+      X[1,element] = modal[1,1]
+      X[2,element] = modal[1,2]
+      X[3,element] = modal[2,1]
+    end
+
+
+    @threaded for element in eachelement(dg, cache)
+      cell_id = cache.elements.cell_ids[element]
+      network_input = Array{Float64}(undef, 15)
+
+      network_input[1] = X[1,element]
+      network_input[2] = X[2,element]
+      network_input[3] = X[3,element]
+
+      #dirshuffle = collect(1:4)
+      #dirshuffle = shuffle!(dirshuffle)
```

Comment:
Commented code: please either remove, uncomment, or add a note explaining why commented code should be retained.